docs/src/quickstart.md

```diff
@@ -5,7 +5,7 @@ Because CUDA is the most popular GPU programming environment, we can use it as a
 reference for defining terminology in KA. A *workgroup* is called a block in
 NVIDIA CUDA and designates a group of threads acting in parallel, preferably
 in lockstep. For the GPU, the workgroup size is typically around 256, while for the CPU,
-it is usually less than or equal to the number of CPU cores. An *ndrange* is
+it is usually a multiply of the natural vector-width. An *ndrange* is
```

Comment:
```suggestion
it is usually a multiple of the natural vector-width. An *ndrange* is
```
This is actually news to me, almost all the examples use 256 threads and _4 cores_. We should change those.

But stupid questiona(s): why don't we use cores? Also: what is a vector-width?

Reply:
I don't know where the 16 snuck in from. We default to 1024 when none is specified.

https://github.com/JuliaGPU/KernelAbstractions.jl/blob/732fc8450637899db7be318e43b36ac05321fa33/src/cpu.jl#L104

It's actually the opposite way around length(ndrange)/length(wkgrp) should be approx nthreads (or multiple thereof since you want to allow for some work balancing)

Reply:
Generally speaking each workgroup is processed by a single task, so you want to balance the overhead of creating tasks.

This tradeoff might change with https://github.com/JuliaGPU/KernelAbstractions.jl/issues/330#issuecomment-1317625382 but I am not sure how to implement that transform yet